### DISCUSSION TITLE
Using freezed in a package: do I need to commit my generated files?

### DISCUSSION THREAD

**Discussion by JesseRiemens** [rrousselGit/freezed#1086] [Q&A] [answered]
I am using a flutter app (say `foo`), which uses a dependency (say `bar`, my own package), which uses `freezed` for some models. 
Now when testing the dependency, all goes well. All files are generated nicely. I `.gitignore` the `.freezed.dart` files, and commit my source. 

Now my app uses `freezed` as well, so I run code generation. This however does not seem to generate code for `bar` as well; I get warnings while generating:
```
[WARNING] build_resolvers:transitive_digests on package:bar/bar.dart:
Unable to read asset, could not compute transitive deps: bar|lib/src/bar.freezed.dart

This may cause less efficient builds, see the following doc for help:
https://github.com/dart-lang/build/blob/master/docs/faq.md#unable-to-read-asset-could-not-compute-transitive-deps
```
And when I `flutter run` my app, I get a lot of errors like this:
```
ERROR: ../../.pub-cache/git/bar-b71365448f1f99b40e1e1fabfa25ed4d21c6dcfe/lib/src/bar.dart:3:6: Error: Error when reading '../../.pub-cache/git/bar-b71365448f1f99b40e1e1fabfa25ed4d21c6dcfe/lib/src/bar.freezed.dart': No such file or directory
ERROR: part 'bar.freezed.dart';
ERROR:      ^
```
Which tells me it was not generated properly for the dependency. 

Am I doing something wrong, or do I just have to commit my generated files?

I'm guessing it has something to do with [this part in build.yaml in `freezed` package:](https://github.com/rrousselGit/freezed/blob/f905fb0dd8e516b7e2fd2780824d2fb43a4c4d98/packages/freezed/build.yaml#L24)
```yml
builders:
  freezed:
    import: "package:freezed/builder.dart"
    builder_factories: ["freezed"]
    build_extensions: { ".dart": [".freezed.dart"] }
    auto_apply: dependents
    build_to: source      # If this would say cache, would it perhaps work?
    runs_before: ["json_serializable|json_serializable"]
```

Any help is appreciated!

**Comment by rrousselGit** [comment]
You don't need to. That's up to you.

On the other hand, if you want to publish a package, the generated files need to be published too. As such:
- If you're using pub.dev to install your package, you need to publish those generated files.
- If you're using a path dependency, then it's fine if the files are gitignored as long as they are present.
- If you're using a git dependency, you'll have to commit those files got `pub get` to obtain them.

**Reply by JesseRiemens** [reply]
Thanks for the clarification, I should have specified I am using git and pub dependencies. I guess I'll have to check them in. 

Is this common for all generators, or just for this one, since it generates the files to source instead of cache?

**Reply by rrousselGit** [reply, marked as answer]
You only need to check them in for git dependencies.
For pub dependencies, that's not necessary. But you may have to update your .pubignore

That's not specific to Freezed
